### DISCUSSION TITLE
How to use linetypes?

### DISCUSSION THREAD

**Discussion by Cyberduke01** [mozman/ezdxf#468] [Q&A] [answered]
I am trying to change the different type of lines to draw. I followed the tutorial on adding the different lines seen here:

https://ezdxf.readthedocs.io/en/stable/tutorials/linetypes.html

But when I somehow not finding how to change the type of the line I am actually drawing, here is an example of what I am trying to achieve:
msp.add_line(start,end, dxfattribs={linetype: 3})

**Comment by chibai** [comment]
This is not how to ask question.~
1. Please show the full code, and save to dxf file.
2. Then upload the dxf file(compress to zip) to prove your concern.

**Reply by Cyberduke01** [reply]
Hi, please find that here: as shown it is still just a solid line. 

`
import ezdxf

doc = ezdxf.new('R2010')  
msp = doc.modelspace() 

msp.add_line((0,0), (10,10),dxfattribs={"linetype": "DOTTED"})

doc.saveas("TEST.dxf")
`

[TEST123.zip](https://github.com/mozman/ezdxf/files/6654275/TEST123.zip)

**Comment by mozman** [comment, marked as answer]
Most DXF resources are referenced by their name like `... dxfattribs={"linetype": "DOTTED}` or `entity.dxf.linetype = "DOTTED"`, for more information about the common graphical DXF attributes see the [docs](https://ezdxf.readthedocs.io/en/stable/dxfentities/dxfgfx.html#common-graphical-dxf-attributes).

**Reply by Cyberduke01** [reply]
Please see my answer to the above post, that still shows my problem.

**Reply by mozman** [reply]
Resources are not setup by default, see see docs for [ezdxf.new()](https://ezdxf.mozman.at/docs/drawing/management.html#ezdxf.new).

**Reply by Cyberduke01** [reply]
Solution: the word "dotted" doesn't work, when changed to something else like "Center" it worked. 

Problem solved, thanks for the help

**Reply by ryangoss** [reply]
Just to follow up since I had similar question.

I don't think it's correct to say that "dotted" doesn't work. As @mozman pointed out you need to add the resources you are going to use, the key is the definition of `linetypes` and there is a good tutorial for [linetypes](https://ezdxf.mozman.at/docs/tutorials/linetypes.html?highlight=my_line_types)
